ruby_basics/11_nested_collections/exercises/nested_hash_exercises.rb

```diff
@@ -0,0 +1,32 @@
+def add_information_about_language(languages, language_name, info_key, info_value)
+  # Take languages (a hash with language names as keys and information about
+  # each language in a nested hash as the corresponding value) and add the
+  # key/value pair info_key/info_value to the nested hash of language_name,
+  # then return the updated languages hash
+end
+
+def add_language(languages, language_name, language_info_value)
+  # Take languages and add the key/value pair language_name/language_info_value
+  # to it, then return languages
+end
+
+def delete_information_about_language(languages, language_name, info_key)
+  # Take languages and delete the key/value pair with key info_key from
+  # language_name, then return languages
+end
+
+def delete_language(languages, language_name)
+  # Take languages and delete the language_name key/value pair, then return
+  # languages
+end
+
+def delete_beautiful_languages(languages)
+  # Take languages and delete any language that has the key/value pair
+  # { is_beautiful => true } listed in its information, then return languages
+end
+
+def find_language_facts(languages, fact_index)
+  # Take languages (now with additional facts added to each language with the
+  # key :facts and value of an array of strings) and return an array containing
+  # the fact that each language has at fact_index of its facts array
+end
```

Comment:
Similar to the last test in the nested array file, since this changes examples, I think it could be good to include an example of `languages` with the `facts`. I think students would appreciate being able to see an example (similar to lines 6-10) in the nested array file. What do you think of this idea?

You mention this being a bit of a weak example. If the goal is to have the students iterate over the nested array, I think it is fine (although it embarrassingly took me a bit to solve) . However, if you are open to the idea of having the students just accessing the facts, then what do you think of adding `language` as a parameter. Then when `:ruby` is provided as an argument, only its facts will be returned`['was made for programmer happiness', 'its code is beautiful']`.

If you want to use the `fact_index`, you could also have students use a default parameter = 0. Then it will always return the first fact, unless another number is provided.

Reply:
Agree on adding an example of the new structure in the instructions.

My thinking was that this exercise should make sure that students know how to access an element from an array within a hash, rather than just returning the whole array, but if you think that direction is better I'm happy to go in that direction.

Reply:
Yes, you are right, students should know how to access a value in that `facts` array, so what do you think the last portion of my idea - using the `fact_index` with a default parameter? Then they will have to access one of the values in the `facts` array. 

```
def find_language_facts (languages, language, fact_index = 0)
  languages[language][:facts][fact_index]
end
```

`find_language_facts(languages, :ruby)` would return `'was made for programmer happiness'`
`find_language_facts(languages, :ruby, 1)` would return `'its code is beautiful'`
`find_language_facts(languages, :ruby, 4)` would return `nil`

That is unless you really want them to have to iterate through the nested hash.

Reply:
Example added in 583611f and exercise refactor done in 24bd935